### PR TITLE
Add sphinxcontrib-trio as a Sphinx extension.

### LINKED ISSUE #793: Use sphinxcontrib-trio to document coroutines
It looks like https://sphinxcontrib-trio.readthedocs.io/ could help autodoc'ing our coroutines nicer.

### PR DESCRIPTION
**Pull request by cmarqu**
Fixes #793.
This alone only adds "_for ... in_" before coroutines in rendered output but for the future enables better (auto-)documentation as described in its docs.

### CODE REVIEW

**Comment by imphil** [comment]
Please don't copy the code in, you can add this extension to `documentation/requirements.txt` to be installed into the virtual environment during docs build.

File: documentation/source/conf.py [hunk 1]
```diff
@@ -141,7 +142,7 @@
 # Add any paths that contain custom static files (such as style sheets) here,
 # relative to this directory. They are copied after the builtin static files,
 # so a file named "default.css" will overwrite the builtin "default.css".
-html_static_path = ['_static']
+#html_static_path = ['_static']
```
Comment (imphil) [inline documentation/source/conf.py]: This change doesn't seem to be related to this PR? If it isn't please remove it from this commit and/or open a new PR.

Comment (cmarqu) [reply documentation/source/conf.py]: Alright,  should be fine now.

**Comment by imphil** [comment]
Thanks, looks good!
